python/ray/data/tests/test_stats.py

```diff
@@ -133,6 +135,76 @@ def test_dataset_stats_sort(ray_start_regular_shared):
     assert "sort_reduce" in stats, stats
 
 
+@patch("ray.data._internal.plan.logger")
+def test_dataset_stats_autolog(mock_logger, ray_start_regular_shared):
+    """Tests functionality to autolog stats after each stage execution,
+    enabled via the `DatasetContext.enable_auto_log_stats` parameter."""
+    context = DatasetContext.get_current()
+    context.enable_auto_log_stats = True
+    ds = ray.data.range(1000, parallelism=10)
+    # Check logged stats after each stage
+    ds = ds.map_batches(lambda x: x)
+    logger_args, logger_kwargs = mock_logger.info.call_args
+    assert (
+        canonicalize(logger_args[0])
+        == """Stage Z read: N/N blocks executed in T, N/N blocks split from parent
```

Comment:
hmm shouldn't it be `read->map_batches`?

Reply:
you're right, i think this was a result of me using `self._snapshot_stats` instead of `stats`, as you recommended above in another comment. i have updated both code sections to reflect this correction, thanks